chunky/src/java/se/llbit/chunky/resources/SignTexture.java

```diff
@@ -36,74 +38,84 @@
     hh = v1 - v0;
   }
 
-  private final Texture texture;
   private final Texture signTexture;
+  private final PalettizedBitmapImage textColor;
+  private final BinaryBitmapImage textMask;
 
   public SignTexture(JsonArray[] text, Texture signTexture) {
     this.signTexture = signTexture;
     int ymargin = 1;
     int lineHeight = 10;
     int width = 96;
     int height = 48;
-    BitmapImage img = new BitmapImage(width, height);
-    int[] data = img.data;
     int ystart = ymargin;
-    for (JsonArray line : text) {
-      if (line.isEmpty()) {
-        ystart += lineHeight;
-        continue;
+    boolean allEmpty = true;
+    for(JsonArray line : text) {
+      if(!line.isEmpty()) {
```

Comment:
To catch some some (probably rare) cases:

```suggestion
      if(!line.trim().isEmpty()) {
```

Reply:
line is a JsonArray so it was a bit more involved but that good